lib/mastermind/score_filter_ai.rb

```diff
@@ -3,19 +3,15 @@
 
 module Mastermind
   class ScoreFilterAi
-    def initialize
-      @initial_guess = true
-      @set_of_guesses = Guess::ALL_PERMUTATIONS
-      @current_guess = "RRGG".split('')
+    def initialize(set_of_guesses = Guess::ALL_PERMUTATIONS)
+      @set_of_guesses = set_of_guesses
+      @current_guess = @set_of_guesses.to_a.sample
       @code_score_calc = Mastermind::CodeScoreCalculator.new
     end
 
     def next_guess(feedback)
```

Comment:
I have left the mutation of the set_of_guesses in the class but it's visible in this method. Also tidied up the method so it checks the feedback to work out if it's the initial guess, so the variable could be removed.